pkg/health/server/server.go

```diff
@@ -417,7 +420,46 @@ func NewServer(config Config) (*Server, error) {
 	server.Client = cl
 	server.Server = *server.newServer(config.HealthAPISpec)
 
-	server.httpPathServer = responder.NewServer(config.HTTPPathPort)
+	server.httpPathServer = responder.NewServers(getAddresses(), config.HTTPPathPort)
 
 	return server, nil
 }
+
+// Get internal node ipv4/ipv6 addresses based on config enabled.
+// If it fails to get either of internal node address, it returns "0.0.0.0" if ipv4 or "::" if ipv6.
+// If it fails to get both, returns empty string array
+func getAddresses() []string {
+	var (
+		addresses = make([]string, 0, 2)
+		ipv4      net.IP
+		ipv6      net.IP
+	)
+	// listen on all interfaces and all families in case of external-workloads
+	if option.Config.JoinCluster {
+		return []string{""}
+	}
+
+	if option.Config.EnableIPv4 {
+		if ipv4 = node.GetInternalIPv4(); ipv4 != nil {
+			addresses = append(addresses, ipv4.String())
+		}
```

Comment:
You can simply use a `else` here to account for the case `ipv4 == nil`, so to avoid the `if len(addresses) != 0` later.